sound/soc/sof/core.c

```diff
@@ -365,6 +365,20 @@ int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 	if (sof_core_debug)
 		dev_info(dev, "sof_debug value: %#x\n", sof_core_debug);
 
+	if (sof_debug_check_flag(SOF_DBG_DSPLESS_MODE)) {
+		if (plat_data->desc->dspless_mode_supported) {
+			if (sof_debug_check_flag(SOF_DBG_FORCE_NOCODEC)) {
```

Comment:
can we combine these 2 inot 1 condition with && and the else can have a single info saying "DSPless mode sectiond ignored"? That will reduce the indendation a bit